maayanlab_bioinformatics/plotting/bridge.py

```diff
@@ -16,13 +16,17 @@ def bridge_plot(select: pd.Series):
   ```
 
   :param select: (pd.Series) selection of hits (i.e. `df['gene'] == 'my_target'`)
+  :param weights: (pd.Series) optional weights for each gene
   :return: (Tuple[np.array, np.array]) x and y arrays ready to be plotted.
   '''
-  up = (~select).sum() / select.shape[0]
-  dn = -(1-up)
+  if weights is None:
+    weights = pd.Series(np.ones(select.shape[0]), index=select.index)
+  max_es = weights[select].abs().sum() # maximum enrichment score if we were to hit everything
+  up = select * weights / max_es # go up by normalized weight on each hit
+  dn = - (1 - select) / (~select).sum() # go down uniformly so we hit 0
```

Comment:
![image](https://user-images.githubusercontent.com/12749033/139613730-b69cb48d-44ff-4297-bc2c-9402ee4f37e9.png)

Reply:
This slide from our past GSEA lecture might help...

Reply:
In the weighted case N is the same, while G is likely the sum of all weights (?)

Reply:
I think we need to decide which paper we'll follow, since I believe this one is different..?

From https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1239896/ in the appendix for Enrichment Score ES(S).

There is this equation:
![image](https://user-images.githubusercontent.com/1341861/139681628-667e38ef-9d7f-4a18-b2f4-81073ec1faa2.png)
![image](https://user-images.githubusercontent.com/1341861/139681729-097369ec-afd4-4782-a2f6-5f8b6020a5dd.png)

ES=P_hit - P_miss

```python
  # weights corresponds to r_j
  if weights is None: weights = pd.Series(np.ones(select.shape[0]), index=select.index)
  # corresponds to N_r
  max_es = weights[select].abs().sum()
  # corresponds to P_hit
  up = select * weights / max_es
  # corresponds to -P_miss with a modification which only applies if weights are < 0
  # up.sum() will be 1 if weights are all positive corresponding to P_miss. (~select).sum() = N-N_h
  dn = - (1 - select) * up.sum() / (~select).sum()
  x = np.arange(select.shape[0]+1)
  y = np.concatenate([
    np.zeros(1),
    np.cumsum(up + dn),
  ])
```

Reply:
I just added a function which follows the paper in implementation and with notation for clarity:
https://github.com/MaayanLab/maayanlab-bioinformatics/blob/fdafe1afde57329d1055caae1c0ba28ac2510986/maayanlab_bioinformatics/enrichment/gsea2015.py#L4-L23

Reply:
After some additional investigation, I found another origin version from 2003: https://pubmed.ncbi.nlm.nih.gov/12808457/

The original author is also on the 2005 version. But notably the 2003 version seems to be aligned with what you referenced @AviMaayan  I can add an additional function: GSEA2003

Reply:
This has been added:

https://github.com/MaayanLab/maayanlab-bioinformatics/blob/04917660b0eda9ed1c278595b18dd96cc87be720/maayanlab_bioinformatics/enrichment/gsea2005.py#L4-L23

Also I've updated https://gist.github.com/u8sand/a132a887fb86094bed759d3dc86b5629 with these new implementations for comparison.

Notably GSEA2003 scale is somewhat odd, judging by some of the output it's possible I have some kind of mistake.